sql/files/defaultdata/r/run

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+# R compile wrapper-script for 'compile.sh'.
+# See that script for syntax and more info.
+#
+# This script does not actually "compile" the source, but writes a
+# shell script that will function as the executable: when called, it
+# will execute the source with the correct interpreter syntax, thus
+# allowing this interpreted source to be used transparantly as if it
+# was compiled to a standalone binary.
+#
+# NOTICE: when this compiler script is used with USE_CHROOT on, the
+# chroot requires r to be installed
+
+DEST="$1" ; shift
+MEMLIMIT="$1" ; shift
+MAINSOURCE="${ENTRY_POINT:-$1}"
+
+# Report the entry point, so it can be saved, e.g. for later replay:
+if [ -z "$ENTRY_POINT" ]; then
+    echo "Info: detected entry_point: $MAINCLASS"
```

Comment:
`MAINCLASS` is undefined here. I see that this is a copy-paste from a mistake that was already there in other compile scripts for scripted languages, though, so actually my mistake ;-)
It should probably be `MAINSOURCE` instead, or in case of a multifile submissiosn, any other file that the compile script can detect as being the primary (I have no experience with R, so don't know whether this makes sense).

Reply:
Why do we actually test in here wheter ENTRY_POINT is not set before we log it is detected?

Reply:
@aginiewicz How about multifile submissions? Is it possible/desired with R?

Reply:
I think multifile submissions should be possible with R, because you can use source to load other script.

Reply:
Does R in that case require you to specify the main file first? Or *only* specify the first file? Or does it not matter

Reply:
Only the first file, other files are sourced on demand

Reply:
@vicrac I think that that means line 46 of runscript needs to change to only contain `$MAINSOURCE` and not `\$@` but I might be wrong. cc @eldering

Reply:
I think I *am* mistaken as we do the same for Python and others so I think this is already fine but Jaap will (hopefully) confirm.

Reply:
The reason to log ENTRY_POINT (which for scripted languages normally is the main source file) is a bit of a technical DOMjudge/Contest API specification issue. Even when the user didn't specify an entry point, we still want to make sure that if we forward the submission to another system (or e.g. rejudge later) that we deterministically use the same entry point as the first time.

Reply:
> @vicrac I think that that means line 46 of runscript needs to change to only contain `$MAINSOURCE` and not `\$@` but I might be wrong. cc @eldering

Indeed. Also if you have a multifile submission and no entry point was specified then it would be good to add some automagic code to detect the main source, that is, the entry point, and log that on this line above.

Reply:
But why don't we do these things (both only passing the main file as well as detecting entry point using a "smarter" way than only picking the first file) for Python2/3? As in this is a direct copy from the python2/3 run-script so I think doing more "smart" things should be done for all these scripts at the same time?

Also regarding `\$@`: it is escaped so these are not the *other* files passed to the runscript, but the arguments passed to the actual program, so that should be fine already

Reply:
> But why don't we do these things (both only passing the main file as well as detecting entry point using a "smarter" way than only picking the first file) for Python2/3? As in this is a direct copy from the python2/3 run-script so I think doing more "smart" things should be done for all these scripts at the same time?

Agreed, we should (besides fixing the bug `MAINCLASS` -> `MAINSOURCE`).

> Also regarding `\$@`: it is escaped so these are not the _other_ files passed to the runscript, but the arguments passed to the actual program, so that should be fine already

Indeed.

Reply:
> > [...]
> 
> Agreed, we should (besides fixing the bug `MAINCLASS` -> `MAINSOURCE`).
> 
> > [...]
> 
> Indeed.

So then I'd say this PR is complete for now, unless of course you or someone finds other things that should be fixed, right? I'll add you as a reviewer and if you approve, let's merge this and then do the changes to the scripts. I think we can copy these changes from the WF branch, where we do more logic: we filter out files with wrong extensions.